standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -0,0 +1,10137 @@
+      long time = System.currentTimeMillis() / 1000;
+      cat.setCreateTime((int) time);
+      cat.setDescription(Warehouse.DEFAULT_CATALOG_COMMENT);
+      ms.createCatalog(cat);
+    }
+  }
+
+  private void createDefaultDB_core(RawStore ms) throws MetaException, InvalidObjectException {
+    try {
+      ms.getDatabase(DEFAULT_CATALOG_NAME, DEFAULT_DATABASE_NAME);
+    } catch (NoSuchObjectException e) {
+      LOG.info("Started creating a default database with name: " + DEFAULT_DATABASE_NAME);
+      Database db = new Database(DEFAULT_DATABASE_NAME, DEFAULT_DATABASE_COMMENT,
+          wh.getDefaultDatabasePath(DEFAULT_DATABASE_NAME, true).toString(), null);
+      db.setOwnerName(HiveMetaStore.PUBLIC);
+      db.setOwnerType(PrincipalType.ROLE);
+      db.setCatalogName(DEFAULT_CATALOG_NAME);
+      long time = System.currentTimeMillis() / 1000;
+      db.setCreateTime((int) time);
+      ms.createDatabase(db);
+      LOG.info("Successfully created a default database with name: " + DEFAULT_DATABASE_NAME);
+    }
+  }
+
+  /**
+   * create default database if it doesn't exist.
+   * <p>
+   * This is a potential contention when HiveServer2 using embedded metastore and Metastore
+   * Server try to concurrently invoke createDefaultDB. If one failed, JDOException was caught
+   * for one more time try, if failed again, simply ignored by warning, which meant another
+   * succeeds.
+   *
+   * @throws MetaException
+   */
+  private void createDefaultDB() throws MetaException {
+    try {
+      RawStore ms = getMS();
+      createDefaultCatalog(ms, wh);
+      createDefaultDB_core(ms);
+    } catch (JDOException e) {
+      LOG.warn("Retrying creating default database after error: " + e.getMessage(), e);
+      try {
+        RawStore ms = getMS();
+        createDefaultCatalog(ms, wh);
+        createDefaultDB_core(ms);
+      } catch (InvalidObjectException | InvalidOperationException e1) {
+        throw new MetaException(e1.getMessage());
+      }
+    } catch (InvalidObjectException | InvalidOperationException e) {
+      throw new MetaException(e.getMessage());
+    }
+  }
+
+  /**
+   * create default roles if they don't exist.
+   * <p>
+   * This is a potential contention when HiveServer2 using embedded metastore and Metastore
+   * Server try to concurrently invoke createDefaultRoles. If one failed, JDOException was caught
+   * for one more time try, if failed again, simply ignored by warning, which meant another
+   * succeeds.
+   *
+   * @throws MetaException
+   */
+  private void createDefaultRoles() throws MetaException {
+    try {
+      createDefaultRoles_core();
+    } catch (JDOException e) {
+      LOG.warn("Retrying creating default roles after error: " + e.getMessage(), e);
+      createDefaultRoles_core();
+    }
+  }
+
+  private void createDefaultRoles_core() throws MetaException {
+
+    RawStore ms = getMS();
+    try {
+      ms.addRole(HiveMetaStore.ADMIN, HiveMetaStore.ADMIN);
```

Comment:
This constant is only used by HMSHandler, so it should be moved to HMSHandler. Please also check other constants in HiveMetaStore that are only used by HMSHandler. Please migrate all.